jbpm-test-coverage/src/test/java/org/jbpm/test/functional/subprocess/SubprocessDoublesTest.java

```diff
@@ -0,0 +1,102 @@
+import org.junit.Test;
+import org.kie.api.event.process.DefaultProcessEventListener;
+import org.kie.api.event.process.ProcessNodeLeftEvent;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.manager.RuntimeEngine;
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemHandler;
+import org.kie.api.runtime.process.WorkItemManager;
+
+import static org.junit.Assert.assertEquals;
+
+public class SubprocessDoublesTest extends JbpmTestCase {
+
+    public SubprocessDoublesTest() {
+        super(true, true);
+    }
+
+    class ExternalTaskWorkItemHandlerTest implements WorkItemHandler {
+
+        public void abortWorkItem(WorkItem workItem, WorkItemManager workItemManager) {
+
+        }
+
+        public void executeWorkItem(WorkItem workItem, WorkItemManager workItemManager) {
+            _lastWorkItemId = workItem.getId();
+        }
+
+        long _lastWorkItemId = 0;
+
+        public long getLastWorkItemId() {
+            return _lastWorkItemId;
+        }
+    }
+
+    @Test
+    public void testSubprocessDoubles() {
+        createRuntimeManager("org/jbpm/test/functional/subprocess/TestSubWorkflow.bpmn2",
+                             "org/jbpm/test/functional/subprocess/TestWorkflow.bpmn2");
+
+        RuntimeEngine runtimeEngine = getRuntimeEngine();
+        KieSession ksession = runtimeEngine.getKieSession();
+        WorkItemManager itemManager = ksession.getWorkItemManager();
+
+        ExternalTaskWorkItemHandlerTest externalTask = new ExternalTaskWorkItemHandlerTest();
+        itemManager.registerWorkItemHandler("GenericTask", externalTask);
+
+        RuleFlowProcessInstance processInstance = (RuleFlowProcessInstance) ksession
+                                                                                    .startProcess("TestProject.TestWorkflow");
+        final AtomicInteger counter = new AtomicInteger(0);
+        ksession.addEventListener(new DefaultProcessEventListener() {
+
+            @Override
+            public void afterNodeLeft(ProcessNodeLeftEvent event) {
+                if (event.getNodeInstance() instanceof ActionNodeInstance) {
+                    counter.incrementAndGet();
+                }
+            }
+        });
+
+        assertProcessInstanceActive(processInstance.getId());
+        assertNodeActive(processInstance.getId(), ksession, "subprocess");
+
+        SubProcessNodeInstance subprocessNode = (SubProcessNodeInstance) processInstance.getNodeInstances().toArray()[0];
+        RuleFlowProcessInstance subprocessInstance = (RuleFlowProcessInstance) ksession
+                                                                                       .getProcessInstance(subprocessNode.getProcessInstanceId());
+
+        // This code simaulates "GET process/instance" rest api call.
+        // Every call of getProcessInstance(id, true) causes one extra-call of firs WorkItem after subprocess.
```

Comment:
I don't understand this comment, what is an extra-call of first work item?

Reply:
it is referred that the the process invokes twices the listener once you get the read only process because it registers twice the same listener